src/pages/index.js

```diff
@@ -72,50 +88,166 @@ newCardPopup.setEventListeners();
   console.log("name:", name, "job:", job); //debug log
   user.setUserInfo({ name, job }); //set user information
+  api.updateUserInfo(profileData.title, profileData.description);
   const updatedUserInfo = user.getUserInfo(); //get updated information and store in a var
   console.log(updatedUserInfo); // debug log
   profilePopup.closePopup(); //close popup
+
+  setTimeout(() => {
+    profileModalSubmitButton.textContent = "Save";
+  }, 1000);
+}
+
+function handleAvatarUpdate(link) {
+  const url = link.url; //extract url from link object and set it to url value
+
+  avatarModalSubmitButton.textContent = "Saving...";
+
+  api
+    .updateProfilePic(url)
+    .then(() => {
+      user.setUserAvatar(url);
+    })
+    .catch((err) => {
+      console.error(`Error caught in HandleAvatarUpdate function ${err}`);
+    });
+
+  setTimeout(() => {
+    avatarModalSubmitButton.textContent = "Save";
+  }, 1000);
+  avatarPopup.closePopup();
 }
 
 function handleAddCardFormSubmit(newCardData) {
-  const name = newCardData.title; //set card name
-  const alt = newCardData.title; //set card alt text
-  const link = newCardData.url; //set card image link
-  console.log(name, alt, link, newCardData); //debug log
-
-  renderCard({ name, link, alt }); //call renderCard to append card to page
-  newCardPopup.closePopup(); //close popup
-  addNewCardValidator.resetForm(); //reset form and disable submit button
+  addNewCardSubmitButton.textContent = "Adding...";
+  api
+    .createNewCard(newCardData.title, newCardData.url)
+    .then((res) => {
+      /* console.log(res); */
+      renderCard(res);
+      newCardPopup.closePopup(); //close popup
+      addNewCardValidator.resetForm(); //reset form and disable submit button
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+
+  setTimeout(() => {
+    addNewCardSubmitButton.textContent = "Add";
+  }, 1000);
+}
+
+function handleCardLikes(cardData) {
+  /*   console.log(cardData);
+  console.log(cardData._like); */
+
+  if (!cardData._like) {
+    //if not liked
+    console.log(cardData._like);
+    api
+      .cardLike(cardData._id)
+      .then(() => {
+        /*    console.log(`Card ${cardData._id} liked`); */
+        cardData.updateLike(true); //call card.js like method
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else if (cardData._like) {
+    //if liked
+    api
+      .cardUnlike(cardData._id)
+      .then(() => {
+        cardData.updateLike(false);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
+
+function handleCardDelete(cardData) {
+  //pass in card data
+
+  deleteConfirm.openPopup(); //open delete modal
+  //close button functionality
+
+  console.log(cardData);
+  deleteModalButton.addEventListener("click", () => {
+    deleteModalButton.textContent = "Deleting...";
+    api
+      .deleteCard(cardData._id)
+      .then(() => {
+        cardData._removeCard(cardData._id); //call card.js delete
```

Comment:
`_removeCard`

* Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.